### ISSUE TITLE
Status issue - can download before setting project as available

### ISSUE THREAD

**Issue by inaod568** [ScilifelabDataCentre/dds_web#817]
Noticed this when testing #815 but I don't think it has to do with that. 
To reproduce:
1.  `dds auth login` with `unituser_1` for example
2. `dds project create`
3. `dds data put`
4. `dds user add --project`, some researchuser that already has an account
5. `dds auth login` with the newly added researchuser
6. `dds data get` with the newly created project

This should not happen since download should only be possible when the project is set to `Available`, which it shouldn't be until manual status change.

**Comment by inaod568** [comment]
@MatthiasZepper Also when I run `dds project status delete` I get 
```
dds_backend   | [2022-02-07 09:41:19,057] errors [WARNING] Invalid status transition
```
but when I do `dds project status display` after that I get this anyway...
```
INFO     Current status of someunit001: Deleted  
```

**Comment by inaod568** [comment]
Fixes with #833
